From ec00a6cd09fbcb05fa7c02df5d45e0447259e6bf Mon Sep 17 00:00:00 2001
From: Mel Gorman <mgorman@suse.de>
Date: Thu, 3 Apr 2014 17:22:15 +0100
Subject: [PATCH] mm: Do not use atomic operations when releasing pages

References: VM/FS Performance
Patch-mainline: v3.16
Git-commit: e3741b506c5088fa8c911bb5884c430f770fb49d

There should be no references to it any more and a parallel mark should
not be reordered against us. Use non-locked varient to clear page active.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 mm/swap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/swap.c b/mm/swap.c
index 2dd4252..ce5a9c2 100644
--- a/mm/swap.c
+++ b/mm/swap.c
@@ -852,7 +852,7 @@ void release_pages(struct page **pages, int nr, bool cold)
 		}
 
 		/* Clear Active bit in case of parallel mark_page_accessed */
-		ClearPageActive(page);
+		__ClearPageActive(page);
 
 		list_add(&page->lru, &pages_to_free);
 	}
